synedrion/src/cggmp21/entities.rs

```diff
@@ -92,40 +110,46 @@ pub struct KeyShareChange<P: SchemeParams, I: Ord> {
     pub(crate) phantom: PhantomData<P>,
 }
 
-/// The result of the Presigning protocol.
-#[derive(Debug, Clone)]
-pub(crate) struct PresigningData<P: SchemeParams, I> {
-    pub(crate) nonce: Scalar, // x-coordinate of $R$
-    /// An additive share of the ephemeral scalar.
-    pub(crate) ephemeral_scalar_share: Secret<Scalar>, // $k_i$
-    /// An additive share of `k * x` where `x` is the secret key.
-    pub(crate) product_share: Secret<Scalar>,
-
-    // Values generated during presigning,
-    // kept in case we need to generate a proof of correctness.
-    pub(crate) product_share_nonreduced: SecretSigned<<P::Paillier as PaillierParams>::Uint>,
-
-    // $K_i$.
-    pub(crate) cap_k: Ciphertext<P::Paillier>,
-
-    // The values for $j$, $j != i$.
-    pub(crate) values: BTreeMap<I, PresigningValues<P>>,
+impl<P: SchemeParams, I: Ord> PublicAuxInfos<P, I> {
+    pub(crate) fn as_map(&self) -> &BTreeMap<I, PublicAuxInfo<P>> {
+        &self.0
+    }
+
+    /// Returns a `PublicAuxInfos` object for the given subset of all parties.
+    pub fn subset(self, parties: &BTreeSet<I>) -> Result<Self, LocalError> {
+        if parties.iter().any(|party| !self.0.contains_key(party)) {
+            return Err(LocalError::new(
+                "`parties` must be a subset of all the available Aux Infos",
+            ));
+        }
+
+        let aux_infos = self.0.into_iter().filter(|(id, _aux)| parties.contains(id)).collect();
+
+        Ok(Self(aux_infos))
+    }
 }
 
-#[derive(Debug, Clone)]
-pub(crate) struct PresigningValues<P: SchemeParams> {
-    pub(crate) hat_beta: SecretSigned<<P::Paillier as PaillierParams>::Uint>,
-    pub(crate) hat_r: Randomizer<P::Paillier>,
-    pub(crate) hat_s: Randomizer<P::Paillier>,
-    pub(crate) cap_k: Ciphertext<P::Paillier>,
-    /// Received $\hat{D}_{i,j}$.
-    pub(crate) hat_cap_d_received: Ciphertext<P::Paillier>,
-    /// Sent $\hat{D}_{j,i}$.
-    pub(crate) hat_cap_d: Ciphertext<P::Paillier>,
-    pub(crate) hat_cap_f: Ciphertext<P::Paillier>,
+impl<P: SchemeParams, I: Clone + Ord + Debug> PublicKeyShares<P, I> {
+    pub(crate) fn as_map(&self) -> &BTreeMap<I, Point> {
+        &self.shares
+    }
 }
 
-impl<P: SchemeParams, I: Clone + Ord + PartialEq + Debug> KeyShare<P, I> {
+impl<P: SchemeParams, I: Clone + Ord + Debug> KeyShare<P, I> {
+    pub(crate) fn new(owner: I, secret: Secret<Scalar>, public_shares: BTreeMap<I, Point>) -> Result<Self, LocalError> {
+        if public_shares.values().sum::<Point>() == Point::IDENTITY {
+            return Err(LocalError::new("Key shares add up to zero"));
```

Comment:
"add up to zero"? Or is it to "Point at Infinity"?

Reply:
Depends, I guess I was thinking of secret shares, which do add up to zero. Public shares add up to infinity.